data/ai/lua/generic_recruit_engine.lua

```diff
@@ -701,28 +703,36 @@ return {
                     recruit_unit.y = target_hex[2]
                 end
 
-                local path, cost = wesnoth.find_path(recruit_unit, enemy_location.x, enemy_location.y, {ignore_units = true})
-                local time_to_enemy = cost / wesnoth.unit_types[recruit_id].max_moves
-                local move_score = 1 / (time_to_enemy * unit_cost^0.5)
+                -- Need non-zero default scores here. The values do not matter as they are normalized later
+                local move_score = 1
+                local offense_score, defense_score = 1, 1
```

Comment:
Hmm, I should put these on the same line.  I dealt with `move_score` differently before, so that's why I have it on a different line.